tests/returntypes/test_spsa_gradient_shot_vec_new.py

```diff
@@ -0,0 +1,1394 @@
+        """Tests correct output shape and evaluation for a tape
+        with prob and expval outputs"""
+        dev = qml.device("default.qubit", wires=2, shots=many_shots_shot_vector)
+        x = 0.543
+        y = -0.654
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RX(x, wires=[0])
+            qml.RY(y, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            qml.expval(qml.PauliZ(0))
+            qml.probs(wires=[0, 1])
+
+        tapes, fn = spsa_grad(
+            tape,
+            approx_order=approx_order,
+            strategy=strategy,
+            validate_params=validate,
+            sampler=coordinate_sampler,
+            num_directions=4,
+            h=h_val,
+            shots=many_shots_shot_vector,
+        )
+        all_res = fn(dev.batch_execute(tapes))
+
+        assert isinstance(all_res, tuple)
+        assert len(all_res) == len(default_shot_vector)
+
+        for res in all_res:
+
+            assert isinstance(res, tuple)
+            assert len(res) == 2
+
+            assert isinstance(res[0], tuple)
+            assert len(res[0]) == 2
+            assert isinstance(res[0][0], numpy.ndarray)
+            assert isinstance(res[0][1], numpy.ndarray)
+
+            assert isinstance(res[1], tuple)
+            assert len(res[1]) == 2
+            assert isinstance(res[1][0], numpy.ndarray)
+            assert isinstance(res[1][1], numpy.ndarray)
+
+            # The coordinate_sampler produces the right evaluation points, but the tape execution
+            # results are averaged instead of added, so that we need to revert the prefactor
+            # 1 / num_params here.
+            res_0 = (2 * res[0][0], 2 * res[0][1])
+            assert np.allclose(res_0[0], -np.sin(x), atol=spsa_shot_vec_tol, rtol=0)
+            assert np.allclose(res_0[1], 0, atol=spsa_shot_vec_tol, rtol=0)
+
+            res_1 = (2 * res[1][0], 2 * res[1][1])
+            assert np.allclose(
+                res_1[0],
+                [
+                    -(np.cos(y / 2) ** 2 * np.sin(x)) / 2,
+                    -(np.sin(x) * np.sin(y / 2) ** 2) / 2,
+                    (np.sin(x) * np.sin(y / 2) ** 2) / 2,
+                    (np.cos(y / 2) ** 2 * np.sin(x)) / 2,
+                ],
+                atol=spsa_shot_vec_tol,
+                rtol=0,
+            )
+            assert np.allclose(
+                res_1[1],
+                [
+                    -(np.cos(x / 2) ** 2 * np.sin(y)) / 2,
+                    (np.cos(x / 2) ** 2 * np.sin(y)) / 2,
+                    (np.sin(x / 2) ** 2 * np.sin(y)) / 2,
+                    -(np.sin(x / 2) ** 2 * np.sin(y)) / 2,
+                ],
+                atol=spsa_shot_vec_tol,
+                rtol=0,
+            )
+
+
+@pytest.mark.parametrize("approx_order", [2])
+@pytest.mark.parametrize("strategy", ["center"])
+@pytest.mark.xfail(reason="TODO: higher-order derivatives with finite shots")
```

Comment:
What is the limitation here? I think it works with finite-diff and shots.

Reply:
It is copied from `test_finite_diff_shot_vec_new`. I suggest to resolve this in a separate PR.

Reply:
Yes great